domain-models-runtime/src/test/java/org/folio/rest/persist/PostgresClientTransactionsIT.java

```diff
@@ -243,4 +259,292 @@ public void test(TestContext context) {
     deleteTransaction(context);
   }
 
+  @Test
+  public void testWithTransactionSuccess(TestContext context) {
+    AtomicInteger open = new AtomicInteger();
+    AtomicInteger active = new AtomicInteger();
+    PostgresClient c1 = postgresClientMonitor(open, active);
+    {
+      Async async = context.async();
+      c1.withTransaction(f -> f
+          .query("INSERT INTO " + fullTable + " VALUES (2, '{\"name\": \"a2\"}');")
+          .execute()
+          .flatMap(res -> f
+              .query("INSERT INTO " + fullTable + " VALUES (3, '{\"name\": \"a3\"}');")
+              .execute().map("inserted 2")
+          )
+      ).onComplete(context.asyncAssertSuccess(res -> {
+        context.assertEquals("inserted 2", res);
+        async.complete();
+      }));
+      async.await();
+    }
+    context.assertEquals(0, open.get());
+    context.assertEquals(0, active.get());
+    c1.select("SELECT jsonb->>'name' FROM " + fullTable + " WHERE ID=2 OR ID=3",
+        context.asyncAssertSuccess(res -> context.assertEquals(2, res.size())));
+  }
+
+  @Test
+  public void testWithTransactionFailure(TestContext context) {
+    AtomicInteger open = new AtomicInteger();
+    AtomicInteger active = new AtomicInteger();
+    PostgresClient c1 = postgresClientMonitor(open, active);
+    {
+      Async async = context.async();
+      c1.withTransaction(f -> f
+          .query("INSERT INTO " + fullTable + " VALUES (4, '{\"name\": \"a4\"}');")
+          .execute()
+          .flatMap(res -> f
+              .query("INSERT INTO " + fullTable + " VALUES (5,")
+              .execute().map("inserted 2")
+          )).onComplete(context.asyncAssertFailure(cause -> {
+        context.assertTrue(cause.getMessage().contains("syntax error at end of input"));
+        async.complete();
+      }));
+      async.await();
+    }
+    context.assertEquals(0, open.get());
+    context.assertEquals(0, active.get());
+    // first one rolled back
+    c1.select("SELECT jsonb->>'name' FROM " + fullTable + " WHERE ID=4",
+        context.asyncAssertSuccess(res -> context.assertEquals(0, res.size())));
+  }
+
+  @Test
+  public void testWithConnectionSuccess(TestContext context) {
+    AtomicInteger open = new AtomicInteger();
+    AtomicInteger active = new AtomicInteger();
+    PostgresClient c1 = postgresClientMonitor(open, active);
+    {
+      Async async = context.async();
+      c1.withConnection(f -> f
+          .query("INSERT INTO " + fullTable + " VALUES (6, '{\"name\": \"a6\"}');")
+          .execute()
+          .flatMap(res -> f
+              .query("INSERT INTO " + fullTable + " VALUES (7, '{\"name\": \"a7\"}');")
+              .execute().map("inserted 2")
+          )).onComplete(context.asyncAssertSuccess(res -> {
+        context.assertEquals("inserted 2", res);
```

Comment:
Good test!

To be sure that each AtomicInteger is actually used:

      context.assertEquals(1, open.get());
      context.assertEquals(1, active.get());

Reply:
ok